frontend/src/components/pages/AdminPage.tsx

```diff
@@ -28,98 +33,117 @@ const pages: Page[] = [
-    );
-  if (error)
-    return (
-      <>
-        <ErrorState />
-      </>
+  const adminData = React.useMemo(() => {
+    const filteredUsers = data?.usersByRole;
+
+    if (!search) return filteredUsers;
+    return filteredUsers.filter(
+      (user: AdminUser) =>
+        `${user.firstName} ${user.lastName}`
+          .toLowerCase()
+          .includes(search.toLowerCase()) ||
+        user.email.toLowerCase().includes(search.toLowerCase()),
     );
+  }, [search, data]);
 
   return (
-    <Flex mx="4">
-      <Box p={0} w="20%">
-        <SideBar pages={pages} />
-      </Box>
-      <Flex direction="column" w="100%" pt={20}>
-        <Text
-          textStyle="header4"
-          color="blue.300"
-          style={{ textAlign: "left" }}
-        >
-          Database
-        </Text>
-        <Box textAlign="right" pr={10}>
-          <AddAdminModal />
-        </Box>
-        <Tabs pr={5}>
-          <TabList>
-            <Tab color={unselectedColor}>Admin</Tab>
-            <Tab color={unselectedColor}>Teachers</Tab>
-          </TabList>
-          <TabPanels>
-            <TabPanel>
-              <AdminUserTable
-                adminUsers={data?.usersByRole?.map((user: AdminUser) =>
-                  getAdminUser(user),
-                )}
+    <Flex margin={0}>
+      <SideBar pages={pages} />
+      <VStack flex="1" align="left" margin="4.5em 2em 0em 2em">
+        <Box>
+          <Text
+            textStyle="header4"
+            color="blue.300"
+            style={{ textAlign: "left" }}
+            marginBottom="0.5em"
+          >
+            Database
+          </Text>
+          <HStack justifyContent="space-between">
+            <InputGroup maxWidth="280px">
+              <Input
+                borderRadius="6px"
+                backgroundColor="grey.100"
+                onChange={(e) => setSearch(e.target.value)}
+                placeholder="Search bar"
               />
-            </TabPanel>
-            <TabPanel>
-              <AdminUserTable
-                adminUsers={data?.usersByRole?.map((user: AdminUser) =>
-                  getAdminUser(user),
-                )}
-              />
-            </TabPanel>
-          </TabPanels>
-        </Tabs>
-      </Flex>
+              <InputRightElement pointerEvents="none" h="full">
+                <SearchBar />
+              </InputRightElement>
+            </InputGroup>
+            <AddAdminModal />
+          </HStack>
+        </Box>
+        <Center flex="1">
+          {loading && <LoadingState />}
+          {error && <ErrorState />}
+          {data && !error && !loading && (
+            <Tabs marginTop={3}>
+              <TabList>
+                <Tab color={unselectedColor}>Admin</Tab>
+                <Tab color={unselectedColor}>Teachers</Tab>
+              </TabList>
+              <TabPanels>
+                <TabPanel>
+                  <AdminUserTable
+                    adminUsers={adminData?.map((user: AdminUser) =>
```

Comment:
i know this part isn't your code, but is there a cleaner way to pass the value of queried admin users? without using the helper function?